mypy/semanal.py

```diff
@@ -953,6 +955,28 @@ def analyze_base_classes(self, defn: ClassDef) -> None:
         if defn.info.is_enum and defn.type_vars:
             self.fail("Enum class cannot be generic", defn)
 
+    def get_base_type_exprs(self, defn: ClassDef) -> List[Expression]:
```

Comment:
Document that this may modify `defn.metaclass` -- maybe also include this information in the method name.